### PR TITLE
Adding windows-hpc-base-image

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

https://github.com/kubernetes/kubernetes/pull/125297 is adding a Windows flavor to the kube-proxy image and there was some concern about basing the image on another image hosted at mcr.microsoft.com.

This PR adds a windows container base image that other images (like the Windows kube-proxy image) can be based on and is functionally equivalent to the image `mcr.microsoft.com/oss/kubernetes/windows-host-process-containers-base-image:v1.0.0` but can be built and hosted in K8s community resouces

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: images/windows-hpc-base/cc0-legalcode.txt [hunk 1]
```diff
@@ -0,0 +1,121 @@
+Creative Commons Legal Code
```
Comment (BenTheElder) [inline images/windows-hpc-base/cc0-legalcode.txt:1]: Shouldn't we just use the repo's ASLv2?

the resulting image on top of the base will be largely ASLv2 anyhow.

Comment (marosset) [reply images/windows-hpc-base/cc0-legalcode.txt:1]: The equivalent image that gets published to MCR was released with the CC0 license - https://github.com/microsoft/windows-host-process-containers-base-image/blob/main/cc0-license.txt

File: images/windows-hpc-base/build-hpc-base-image.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+BUILD_DIR=build
+
+#trap "rm -rf $BUILD_DIR" EXIT
+
+# Create necessary sub-directorctories
+mkdir -p "$BUILD_DIR/layer/Files/Windows/System32/config"
+
+# Create the files that ProcessBaseLayer on Windows validates when unpacking container images
+touch "$BUILD_DIR/layer/Files/Windows/System32/config/DEFAULT"
+echo "" > "$BUILD_DIR/layer/Files/Windows/System32/config/DEFAULT"
```
Comment (puerco) [inline images/windows-hpc-base/build-hpc-base-image.sh:15]: I think the touch in these lines is superfluous. Maybe drop them?

File: images/windows-hpc-base/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+To build the image tarball:
+
+```sh
+make image_build
+```
+
+To publish the image to a registry:
+
+```sh
+REGISTRY=gcr.io/k8s-staging-releng VERSION=v0.1.0 make image_publish
```
Comment (puerco) [inline images/windows-hpc-base/README.md:16]: The image `image_publish` target is missing from the makefile above.

**Review by puerco — COMMENTED** [review]
I started reviewing this yesterday. I have a few comments, one of them is about the license which Ben already mentioned

File: images/windows-hpc-base/build-hpc-base-image.sh (hunk 2 above)
Comment (marosset) [reply images/windows-hpc-base/build-hpc-base-image.sh:15]: I think I can remove the echo actually.
Let me verify and I'll remove one or the other

File: images/windows-hpc-base/README.md (hunk 3 above)
Comment (marosset) [reply images/windows-hpc-base/README.md:16]: that should be image_push :-/ - i'll update

File: images/windows-hpc-base/cc0-legalcode.txt (hunk 1 above)
Comment (BenTheElder) [reply images/windows-hpc-base/cc0-legalcode.txt:1]: I mean we're going to build ASLv2 images with this, and CC0 is NOT on the CNCF approved list so we should not be using any CC0 images.

Are we using CC0 anywhere else ...? 

https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md

Comment (puerco) [reply images/windows-hpc-base/cc0-legalcode.txt:1]: Correct.

Comment (puerco) [reply images/windows-hpc-base/cc0-legalcode.txt:1]: Also (I'm sure you saw mark) but the script and Makefile need the boilerplate repo headers, and the automation will look for the apache license and fail if we use a different one.

**Pull request closed without merging**